src/scenic/simulators/robosuite/simulator.py

```diff
@@ -0,0 +1,566 @@
+            # No tables - use empty arena (just floor)
+            return EmptyArena()
+        
+        # Use MultiTableArena for any number of tables
+        return MultiTableArena(
+            table_offsets=[t.get('position', [0, 0, 0.8]) for t in self.scenic_tables],
+            table_full_sizes=[t.get('size', (0.8, 0.8, 0.05)) for t in self.scenic_tables],
+            has_legs=[True] * len(self.scenic_tables)
+        )
+    
+    def _create_objects(self) -> List:
+        """Create Robosuite objects from Scenic configuration."""
+        mujoco_objects = []
+        
+        for obj_config in self.scenic_objects:
+            obj_type = obj_config['type']
+            factory = OBJECT_FACTORIES.get(obj_type, OBJECT_FACTORIES['Box'])
+            mj_obj = factory(obj_config)
+            mujoco_objects.append(mj_obj)
+        
+        return mujoco_objects
+    
+    def _setup_references(self):
+        """Setup references to simulation objects."""
+        super()._setup_references()
+        self.obj_body_ids = {
+            obj.name: self.sim.model.body_name2id(obj.root_body)
+            for obj in self.mujoco_objects
+        }
+    
+    def _setup_observables(self) -> Dict:
+        """Setup observables for objects."""
+        observables = super()._setup_observables()
+        
+        for obj in self.mujoco_objects:
+            @sensor(modality="object")
+            def obj_pos(obs_cache, name=obj.name):
+                return np.array(self.sim.data.body_xpos[self.obj_body_ids[name]])
+            
+            obj_pos.__name__ = f"{obj.name}_pos"
+            observables[obj_pos.__name__] = Observable(
+                name=obj_pos.__name__,
+                sensor=obj_pos,
+                sampling_rate=self.control_freq
+            )
+        
+        return observables
+    
+    def _reset_internal(self):
+        """Reset environment internals."""
+        super()._reset_internal()
+        
+        # Set positions for all objects
+        for obj_config, mj_obj in zip(self.scenic_objects, self.mujoco_objects):
+            if 'position' in obj_config:
+                pos = obj_config['position']
+                quat = obj_config.get('quaternion', [1, 0, 0, 0])
+                self.sim.data.set_joint_qpos(
+                    mj_obj.joints[0],
+                    np.concatenate([np.array(pos), np.array(quat)])
+                )
+    
+    def reward(self, action=None) -> float:
+        """Compute reward."""
+        return 0.0
+    
+    def _check_success(self) -> bool:
+        """Check task success."""
+        return False
+
+
+class RobosuiteSimulator(Simulator):
+    """Simulator for RoboSuite custom environments."""
+    
+    def __init__(self, render: bool = True, real_time: bool = True, speed: float = 1.0,
+                 env_config: Optional[Dict] = None, controller_config: Optional[Dict] = None,
+                 camera_view: Optional[str] = None, lite_physics: Optional[bool] = None):
```

Comment:
Docstring should explain what these optional arguments do. (See `CarlaSimulator` or `MetaDriveSimulator` for examples.)